orderBook.js

```diff
@@ -0,0 +1,41 @@
+const isMatching = (existingOrder, incomingOrder) => incomingOrder.type === 'buy'
+  ? incomingOrder.price >= existingOrder.price
+  : incomingOrder.price <= existingOrder.price
+
+const isMatchingType = (existingOrder, incomingOrder) =>
+  existingOrder.type !== incomingOrder.type && isMatching(existingOrder, incomingOrder)
+
+
+const isFulfill = (matchingOrder, incomingOrder) => matchingOrder.quantity <= incomingOrder.quantity
```

Comment:
`is` and `has` are for predicate functions. This function returns an object so `fulfillOrder` would be a better name